encodings/dict/src/compute/min_max.rs

```diff
@@ -1,16 +1,135 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: Copyright the Vortex contributors
 
-use vortex_array::compute::{MinMaxKernel, MinMaxKernelAdapter, MinMaxResult, min_max, take};
-use vortex_array::register_kernel;
+use vortex_array::compute::{MinMaxKernel, MinMaxKernelAdapter, MinMaxResult, mask, min_max};
+use vortex_array::{Array as _, ToCanonical, register_kernel};
+use vortex_buffer::BitBufferMut;
+use vortex_dtype::match_each_unsigned_integer_ptype;
 use vortex_error::VortexResult;
+use vortex_mask::Mask;
 
 use crate::{DictArray, DictVTable};
 
 impl MinMaxKernel for DictVTable {
     fn min_max(&self, array: &DictArray) -> VortexResult<Option<MinMaxResult>> {
-        min_max(&take(array.values(), array.codes())?)
+        let codes_validity = array.codes().validity_mask();
+        if codes_validity.all_false() {
+            return Ok(None);
+        }
+
+        let codes_primitive = array.codes().to_primitive();
+        let values_len = array.values().len();
+        match_each_unsigned_integer_ptype!(codes_primitive.ptype(), |P| {
+            // mask() sets values to null where the mask is true, so we start
+            // with a fully-set bit buffer.
+            let mut unreferenced = BitBufferMut::new_set(values_len);
+            for (code_idx, &code) in codes_primitive.as_slice::<P>().iter().enumerate() {
+                if codes_validity.value(code_idx) {
+                    #[allow(clippy::cast_possible_truncation)]
+                    unreferenced.unset(code as usize);
```

Comment:
can we use set_unchecked?

Reply:
I tried to, but I don't think it's public unfortunately.

Reply:
Made it pub